v2/internal/util/randextensions/uuid5.go

```diff
@@ -13,27 +13,38 @@ import (
 // Namespace is the ASOv2 UUIDv5 namespace UUID
 var Namespace = uuid.Must(uuid.Parse("9a329043-7ad7-4b1c-812f-9c7a93d6392a"))
 
-// MakeUniqueResourceString generates a string that uniquely identifies a cluster resource.
-func MakeUniqueResourceString(group string, kind string, namespace string, name string) string {
-	return fmt.Sprintf("%s/%s:%s/%s", group, kind, namespace, name)
+// MakeUniqueResourceScopedString generates a string that uniquely identifies a cluster resource. It includes the
+// following distinguishing parts:
+// * Group
+// * Kind
+// * Namespace
+// * Name
+func MakeUniqueResourceScopedString(gk schema.GroupKind, namespace string, name string) string {
+	return fmt.Sprintf("%s/%s:%s/%s", gk.Group, gk.Kind, namespace, name)
 }
 
-// TODO: Fix name
-// MakeUniqueResourceString generates a string that uniquely identifies a cluster resource.
-func MakeUniqueResourceString2(ownerGK schema.GroupKind, ownerName string, gk schema.GroupKind, namespace string, name string) string {
+// TODO: This method has a bug where it is called with an empty owner gk when the owner is an ARM ID.
```

Comment:
I'm not 100% sure what we should do about this bug. We can fix it (which could cause people to leak resources? Maybe not because our validation will protect us possibly? I need to do some testing...)

Reply:
I filed #4079 to track this in 2.9.0 and fixed this bug but _only for the new resource RoleDefinition_. We can track fixing it for existing resources as part of 2.9.0 with the above bug.